src/components/product/product-extractor.service.ts

```diff
@@ -114,14 +123,63 @@ const parseProductRow =
       };
     }
     assert(sheet.isWritten());
-    const bookName = parseScripture(sheet.bookName(row))[0]!.start.book;
+    /**  Load Scripture References **/
+    let scriptureReferences: ScriptureRangeInput[] = [];
+    let unspecifiedScripture: UnspecifiedScriptureType = null;
+    const totalVersesToTranslate = sheet.totalVerses(row) ?? 0;
+    const bookName = sheet.bookName(row)!;
+    const book = Book.tryFind(bookName);
+    const versesInBook = book ? book.totalVerses : null;
+    const isWholeBook = versesInBook === totalVersesToTranslate;
+
+    book
+      ? isWholeBook
+        ? (scriptureReferences = [
+            {
+              start: book.firstChapter.firstVerse.reference,
+              end: book.lastChapter.lastVerse.reference,
+            },
+          ])
+        : (scriptureReferences = findScriptures(sheet.myNote(row)!))
+      : (scriptureReferences = findScriptures(bookName));
+
+    unspecifiedScripture =
+      scriptureReferences.length > 0 || !book
+        ? null
+        : {
+            book: book.name,
+            totalVerses: totalVersesToTranslate,
+          };
```

Comment:
This seems a bit convoluted. We already know `parseScripture` will give us valid ranges otherwise we wouldn't be here. We can just sum up the count of the verses from the ranges and compare that to the total given.
```ts
const isKnown = sumedVerseCountFromRanges === sheet.totalVerses(row)!;
```
If known, we can just forward the scripture on plainly. Otherwise as unknown portion.
```ts
const scripture = isKnown ? parsedScripture : null;
const unknownPortion = isKnown ? null : { book: parsedScripture[0].start.book, totalVerses };
```

Reply:
Tried to resolve per the suggestion above, but in order to handle the additional check for scripture ranges in the myNotes column of the PnP more logic was needed.